nanoFramework.CoreLibrary/System/SpanByte.cs

```diff
@@ -0,0 +1,172 @@
+                }
+            }
+            else
+            {
+                if ((start != 0) || (length != 0))
+                {
+                    throw new ArgumentOutOfRangeException($"Array is null but start and length are not 0");
+                }
+            }
+
+            _array = array;
+            _start = start;
+            _length = length;
+        }
+
+        /// <summary>
+        /// Gets the element at the specified zero-based index.
+        /// </summary>
+        /// <param name="index">The zero-based index of the element.</param>
+        /// <returns>The element at the specified index.</returns>
+        public byte this[int index]
+        {
+            get
+            {
+                return _array[_start + index];
+            }
+            set
+            {
+                _array[_start + index] = value;
+            }
+        }
+
+        /// <summary>
+        /// Returns an empty System.Span object.
+        /// </summary>
+        public static SpanByte Empty => new SpanByte();
+
+        /// <summary>
+        /// Returns the length of the current span.
+        /// </summary>
+        public int Length => _length;
+
+        /// <summary>
+        /// Returns a value that indicates whether the current System.Span is empty.
+        /// true if the current span is empty; otherwise, false.
+        /// </summary>
+        public bool IsEmpty => Length == 0;
+
+        /// <summary>
+        /// Copies the contents of this System.Span into a destination System.Span.
+        /// </summary>
+        /// <param name="destination"> The destination System.Span object.</param>
+        /// <exception cref="System.ArgumentException">
+        /// destination is shorter than the source System.Span.
+        /// </exception>
+        public void CopyTo(SpanByte destination)
+        {
+            if (destination.Length < _length - _start)
+            {
+                throw new ArgumentException($"Destination too small");
+            }
+
+            for (int i = 0; i < _array.Length; i++)
+            {
+                destination[i] = _array[i];
+            }
+        }
+
+        /// <summary>
+        /// Forms a slice out of the current span that begins at a specified index.
+        /// </summary>
+        /// <param name="start">The index at which to begin the slice.</param>
+        /// <returns>A span that consists of all elements of the current span from start to the end of the span.</returns>
+        /// <exception cref="System.ArgumentOutOfRangeException">start is less than zero or greater than System.Span.Length.</exception>
+        public SpanByte Slice(int start)
+        {
+            if ((start > _length - _start) || (start < 0))
+            {
+                throw new ArgumentOutOfRangeException($"start is less than zero or greater than length");
+            }
+
+            return new SpanByte(_array, _start + start, Length - start - _start);
```

Comment:
I think this should be:
```suggestion
            return new SpanByte(_array, _start + start, _length - start /* This could also be _array.Length - _start - start */);
```